uefi/src/data_types/strs.rs

```diff
@@ -211,10 +214,21 @@ impl CStr16 {
         Self::from_u16_with_nul_unchecked(slice::from_raw_parts(ptr, len + 1))
     }
 
-    /// Creates a C string wrapper from a u16 slice
-    ///
-    /// Since not every u16 value is a valid UCS-2 code point, this function
-    /// must do a bit more validity checking than CStr::from_bytes_with_nul
+    /// Creates a `&CStr16` from a u8 slice, if the slice contains exactly
+    /// one terminating null-byte and all chars are valid UCS-2 chars.
```

Comment:
This implies that the string has an odd length (two-byte characters plus one 0 byte), but below you error out in this case?